File: saved-state-of-custom-view/SavedState.java
import android.os.Parcel;
import android.os.Parcelable;
import android.view.View;

class SavedState extends View.BaseSavedState {

    public SavedState(final Parcel source, final ClassLoader loader) {
        super(source, loader);
    }

    public SavedState(final Parcelable superState) {
        super(superState);
    }

    public static final ClassLoaderCreator<SavedState> CREATOR = new ClassLoaderCreator<SavedState>() {
        @Override
        public SavedState createFromParcel(final Parcel source) {
            return new SavedState(source, null);
        }

        @Override
        public SavedState[] newArray(final int size) {
            return new SavedState[size];
        }

        @Override
        public SavedState createFromParcel(final Parcel source, final ClassLoader loader) {
            return new SavedState(source, loader);
        }
    };
}

File: saved-state-of-custom-view/SavedState.kt
import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.ClassLoaderCreator
import android.view.View

class SavedState : View.BaseSavedState {

    constructor(source: Parcel?, loader: ClassLoader?) : super(source, loader)
    constructor(superState: Parcelable?) : super(superState)

    companion object {

        @JvmField
        val CREATOR: ClassLoaderCreator<SavedState> = object : ClassLoaderCreator<SavedState> {
            override fun createFromParcel(source: Parcel): SavedState = SavedState(source, null)

            override fun newArray(size: Int): Array<SavedState?> = arrayOfNulls(size)

            override fun createFromParcel(source: Parcel, loader: ClassLoader): SavedState =
                SavedState(source, loader)
        }
    }
}